### PR TITLE
Fix issue #2288. Backup fog state and disable before drawing UI's.

### PR DESCRIPTION
**Pull request by giladreich**
Already explained in the issue #2288.
Fixes for Direct3D 9.
Need to have a look at Direct3D 10 and 11 now.

### CODE REVIEW

**Comment by ocornut** [comment]
Feedback:
- Using tabs instead of space
- Not updated both changelog
- Are you sure it needs to be backuped since we already backup a state block? (Need to be teated/confirmed that the state block include the fog state)

**Comment by giladreich** [comment]
> Feedback:
> 
> * Using tabs instead of space
> * Not updated both changelog
> * Are you sure it needs to be backuped since we already backup a state block? (Need to be teated/confirmed that the state block include the fog state)

Done :)

**Comment by ocornut** [comment]
One missing changelog entry and we’re there :) thanks!

File: examples/imgui_impl_dx9.cpp [hunk 1]
```diff
@@ -131,6 +132,7 @@ void ImGui_ImplDX9_RenderDrawData(ImDrawData* draw_data)
     g_pd3dDevice->SetRenderState(D3DRS_DESTBLEND, D3DBLEND_INVSRCALPHA);
     g_pd3dDevice->SetRenderState(D3DRS_SCISSORTESTENABLE, true);
     g_pd3dDevice->SetRenderState(D3DRS_SHADEMODE, D3DSHADE_GOURAUD);
+    g_pd3dDevice->SetRenderState(D3DRS_FOGENABLE, FALSE);
```
Comment (ocornut) [inline examples/imgui_impl_dx9.cpp]: Also notice the surrounding code is using C++-style 'false` and yours is using `FALSE`.

**Pull request closed without merging**